avm/src/main.rs

```diff
@@ -61,69 +64,90 @@ fn main() {
         .arg_from_usage("<AGZ_FILE> 'Specifies the .agz file to deploy'")
       )
     )
+    .subcommand(SubCommand::with_name("daemon")
+      .about("Run an .agz file in daemon mode. Used on deploy within cloud provider VMs.")
+      .arg_from_usage("<APP_ID> 'Specifies the alan app to upgrade'")
+      .arg_from_usage("<AGZ_FILE> 'Specifies the .agz file to deploy'")
+      .arg_from_usage("<DEPLOY_TOKEN> 'Specifies the .agz file to deploy'")
+    )
     .arg_from_usage("[SOURCE] 'Specifies a source ln file to compile and run'");
 
   let matches = app.clone().get_matches();
 
-  match matches.subcommand() {
-    ("run",  Some(matches)) => {
-      let agc_file = matches.value_of("FILE").unwrap();
-      let fp = &format!(
-        "{:}/{:}",
-        env::current_dir().ok().unwrap().to_str().unwrap(),
-        agc_file
-      );
-      exec(&fp, false);
-    },
-    ("compile",  Some(matches)) => {
-      let source_file = matches.value_of("INPUT").unwrap();
-      let dest_file = matches.value_of("OUTPUT").unwrap();
-      std::process::exit(compile(&source_file, &dest_file, false));
-    },
-    ("install",  _) => {
-      let source_file = ".dependencies.ln";
-      if Path::new(source_file).exists() {
-        std::process::exit(compile_and_run(source_file));
-      } else {
-        println!(
-          "{} does not exist. Dependencies can only be installed for {}",
-          source_file, source_file
+  let rt = Builder::new_multi_thread()
+    .enable_time()
+    .enable_io()
+    .build()
+    .unwrap();
```

Comment:
Weird that Tokio lives in here, now, but it makes sense.

Reply:
i usually use [this](https://docs.rs/tokio/1.2.0/tokio/attr.main.html) ;)

Reply:
I also prefer that, usually, but in this case the macro does not make it obvious that `time` and `io` are on so I prefer the specificity of the builder (I once wasted like 30min because I was trying to do `io` within the runtime and the error was not very helpful then

Reply:
er, it calls `enable_all` so it *should* enable `io` (maybe you didn't have the feature enabled when you were doing that?)

It's nbd to manually build the runtime and we may need to do it anyways since it provides more flexibility, I was just throwing out there that it's an option that makes it so that the entire program can be `async`

Reply:
Got it, that makes sense. Maybe I was using a builder and it was a me issue and not a tokio issue. Will leave this as-in case we need to configure something down the line